jetty-server/src/main/java/org/eclipse/jetty/server/ServletAttributes.java

```diff
@@ -0,0 +1,83 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.util.Set;
+
+import org.eclipse.jetty.util.Attributes;
+import org.eclipse.jetty.util.AttributesMap;
+
+public class ServletAttributes implements Attributes
+{
+    private final Attributes _attributes = new AttributesMap();
+    private AsyncAttributes _asyncAttributes;
+
+    public void setAsyncAttributes(AsyncAttributes attributes)
+    {
+        _asyncAttributes = attributes;
+    }
+
+    @Override
+    public void removeAttribute(String name)
+    {
+        if (_asyncAttributes == null)
+            _attributes.removeAttribute(name);
+        else
+            _asyncAttributes.removeAttribute(name);
```

Comment:
I think it is clearer as:

```suggestion
        (_asyncAttributes == null ? _asyncAttributes : _attributes).removeAttribute(name);
```

or maybe even with a private method:
```java
    attributes().removeAttribute(name);
```
same for other methods below.